diff --git a/Godeps/Godeps.json b/Godeps/Godeps.json
index 16ed5c5..be6d1e0 100644
--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -5,7 +5,7 @@
 	"Deps": [
 		{
 			"ImportPath": "gitlab.com/qrhelper/qrhelperdatastructs",
-			"Rev": "d6fceefb356f82ea098faee3fa97621bdea53b7e"
+			"Rev": "ba5dfe3d87a1cb6bea9dd1d362f2d7407d8756ea"
 		},
 		{
 			"ImportPath": "github.com/Sirupsen/logrus",
diff --git a/main_test.go b/main_test.go
index 3879591..87399ba 100644
--- a/main_test.go
+++ b/main_test.go
@@ -15,12 +15,12 @@ import (
 
 var ()
 
-var resource1 = datastructs.Resource{Uuid: "333c1dd2-d454-11e6-a110-b34e9f2c654a", Description: "yahoo", Action: "forward", Address: "https://www.yahoo.com"}
-var resource2 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d623", Description: "yahoo", Action: "forward", Address: "https://www.google.com"}
-var resource3 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d624", Description: "redirect", Action: "redirect", Address: "/test"}
-var resource4 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d625", Description: "forward", Action: "forward", Address: "/test"}
-var resource5 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d626", Description: "S3 redirect", Action: "s3serve", Address: "/s3redirect"}
-var resource6 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d627", Description: "S3 forward", Action: "s3redirect", Address: "/s3forward"}
+var resource1 = datastructs.Resource{Uuid: "333c1dd2-d454-11e6-a110-b34e9f2c654a", Description: "yahoo", Protected: "false", Action: "forward", Address: "https://www.yahoo.com", AccessCount: 0}
+var resource2 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d623", Description: "yahoo", Protected: "false", Action: "forward", Address: "https://www.google.com", AccessCount: 0}
+var resource3 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d624", Description: "redirect", Protected: "false", Action: "redirect", Address: "/test", AccessCount: 0}
+var resource4 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d625", Description: "forward", Protected: "false", Action: "forward", Address: "/test", AccessCount: 0}
+var resource5 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d626", Description: "S3 redirect", Protected: "false", Action: "s3serve", Address: "/s3redirect", AccessCount: 0}
+var resource6 = datastructs.Resource{Uuid: "444edd7c-d454-11e6-92b9-374c2fc3d627", Description: "S3 forward", Protected: "false", Action: "s3redirect", Address: "/s3forward", AccessCount: 0}
 
 var Server dbtest.DBServer
 
diff --git a/mongocall_test.go b/mongocall_test.go
index 4cd5448..ce1fbff 100644
--- a/mongocall_test.go
+++ b/mongocall_test.go
@@ -22,6 +22,7 @@ func testResource(r1 datastructs.Resource, r2 datastructs.Resource) bool {
 	if r1.Action == r2.Action &&
 		r1.Address == r2.Address &&
 		r1.Description == r2.Description &&
+		r1.Protected == r2.Protected &&
 		r1.Uuid == r2.Uuid {
 		resflag = true
 	} else {
diff --git a/mongofunctions.go b/mongofunctions.go
index 552599c..36aa81e 100644
--- a/mongofunctions.go
+++ b/mongofunctions.go
@@ -11,22 +11,21 @@ import (
 //FetchResource - fetch resource
 func FetchResource(resourceid string) datastructs.Resource {
 
+	// change := mgo.Change{
+	// 	Update:    bson.M{"$inc": bson.M{"n": 1}},
+	// 	ReturnNew: true,
+	// }
+
 	c := MgoSession.DB(MongoDBDatabase).C("res")
 	result := datastructs.Resource{}
+
 	err = c.Find(bson.M{"uuid": resourceid}).One(&result)
+	//ls err := result.Apply(change, &result)
 
 	if err != nil {
 		log.Print(err)
 		//panic(err)
 	}
-
-	// // Update
-	colQuerier := bson.M{"uuid": resourceid}
-	change := bson.M{"$inc": bson.M{"accesscount": 1}}
-	err = c.Update(colQuerier, change)
-	if err != nil {
-		panic(err)
-	}
 	log.Println("leaving FetchResource")
 	return result
 }
diff --git a/vendor/gitlab.com/qrhelper/qrhelperdatastructs/datastructs.go b/vendor/gitlab.com/qrhelper/qrhelperdatastructs/datastructs.go
index 3669702..7743e94 100644
--- a/vendor/gitlab.com/qrhelper/qrhelperdatastructs/datastructs.go
+++ b/vendor/gitlab.com/qrhelper/qrhelperdatastructs/datastructs.go
@@ -34,7 +34,7 @@ type Resource struct {
 	Protected   string        `json:"Protected"`
 	Action      string        `json:"Action"`
 	Address     string        `json:"Address"`
-	Accesscount int64         `json:"Accesscount"`
+	AccessCount int32         `json:"AccessCount"`
 }
 
 type Resources []Resource
